Enable shadowrealm testing for dom events api

rwaldron opened the pull request.

rwaldron committed:
Enable shadowrealm testing for dom events api

annevk commented:
Let's block on this as per https://github.com/web-platform-tests/wpt/pull/41985.

ptomato committed:
Add ShadowRealm-specific dom/events/ tests avoiding document.createEvent

The non-ShadowRealm version of these tests rely on document.createEvent()
which is not exposed in ShadowRealms. Add separate tests for ShadowRealm
scopes that use the Event constructor instead of document.createEvent().

For the cancelBubble test, we cannot actually test the bubbling
behaviour without a DOM, but the property still must exist.

Also, make sure to expose the export on globalThis in the support script.

ptomato committed:
Add event propagation test on an EventTarget

This adds a copy of the test in Event-propagation.html, but on a plain
EventTarget instead of a DOM node. The difference is that EventTarget
doesn't have bubbling/capture behaviour, so the expectations after
stopPropagation() and cancelBubble=true are different.

This test can be run in every global scope where EventTarget is available,
not just in ShadowRealm.

ptomato committed:
Minor improvements to a few dom/events/ tests

Add assertion messages, fix typo. These are here because I've added
additional variants of these tests for ShadowRealm and have made the same
changes in the new tests.

ptomato commented:
@annevk I believe this can be unblocked for the same reason as #41965. I've rebased it and added some more tests for functionality in ShadowRealms to match the DOM-dependent tests where we can't just add the "run in ShadowRealm" metadata flag.

annevk reviewed (commented):
Why add a bunch of copies of `.html` files instead of turning those into `.any.js` tests? (And ideally turning them into `.any.js` files would happen in a separate PR.)

ptomato commented:
@annevk Sure, I can do that. I did it this way initially, because I assumed we'd still want to keep the existing test coverage of events being created with `document.createEvent()` / `Event.prototype.initEvent()` and dispatched on DOM elements. Should I still keep those as separate `.html` files or do we not care much about that coverage?

annevk commented:
Tests that cannot be generalized should be kept. They could potentially be moved to a `window.js` file to avoid some boilerplate.

ptomato commented:
This is done in #49211.
